md-dao/src/main/java/com/wipro/ats/bdre/md/dao/ProcessDAO.java

```diff
@@ -702,5 +725,155 @@ else  if (childProcess.getProcessType().getProcessTypeId() == 36){
+                    (userRolesNameList.contains(processCreater)&&readList.contains(process.getPermissionTypeByGroupAccessId().getPermissionTypeId()))
+                    || (!userRolesNameList.contains(processCreater)&&readList.contains(process.getPermissionTypeByOthersAccessId().getPermissionTypeId()))
+                    )
+            {
+                return "ACCESS GRANTED";
+            }
+            else
+            {
+                LOGGER.info("user read");
+                throw new SecurityException("ACCESS DENIED");
+            }
+            case "execute": if (executeList.contains(process.getPermissionTypeByUserAccessId().getPermissionTypeId())||
+                    (userRolesNameList.contains(processCreater)&&executeList.contains(process.getPermissionTypeByGroupAccessId().getPermissionTypeId()))
+                    || (!userRolesNameList.contains(processCreater)&&executeList.contains(process.getPermissionTypeByOthersAccessId().getPermissionTypeId()))
+                    )
+            {
+                return "ACCESS GRANTED";
+            }
+            else
+            {
+                LOGGER.info("user execute");
+                throw new SecurityException("ACCESS DENIED");
+            }
+            default: {LOGGER.info("no operation");
+                throw new SecurityException("no operation");
+            }
+
+        }
+
+
+    }
+    else{
+    if (userRolesNameList.contains(processCreater))
+    {
+        switch (action){
+            case "write": if (writeList.contains(process.getPermissionTypeByGroupAccessId().getPermissionTypeId()))
+                               {
+                                   return "ACCESS GRANTED";
+                               }
+                           else
+                          {
+                              LOGGER.info("group write");
+                              throw new SecurityException("ACCESS DENIED");
+                          }
+            case "read": if (readList.contains(process.getPermissionTypeByGroupAccessId().getPermissionTypeId()))
+                               {
+                                   return "ACCESS GRANTED";
+                               }
+                           else
+                          {
+                              LOGGER.info("group read");
+                              throw new SecurityException("ACCESS DENIED");
+                          }
+            case "execute": if (executeList.contains(process.getPermissionTypeByGroupAccessId().getPermissionTypeId()))
+                            {
+                                return "ACCESS GRANTED";
+                            }
+                            else
+                            {
+                                LOGGER.info("group execute");
+                                throw new SecurityException("ACCESS DENIED");
+                            }
+            default: {LOGGER.info("no operation");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this nested code block into a method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1199)